javalin/src/main/java/io/javalin/plugin/tracing/JaegerPlugin.kt

```diff
@@ -0,0 +1,37 @@
+package io.javalin.plugin.tracing
+
+import io.jaegertracing.Configuration
+import io.jaegertracing.internal.JaegerTracer
+import io.jaegertracing.internal.samplers.ProbabilisticSampler
+import io.javalin.Javalin
+import io.javalin.core.plugin.Plugin
+import io.opentracing.util.GlobalTracer
+
+class JaegerPlugin @JvmOverloads constructor(
```

Comment:
this is the main plugin that adds the before and after middleware

Reply:
This needs to check if the dependency is present. I was going to say you can look at how the MicroMeter plugin does it, but it seems they removed the check. It will look something like 

```
 Util.ensureDependencyPresent(OptionalDependency.JAEGER)
```

Which should be called before attempting to load any Jaeger class. This way Javalin will print a block of maven/gradle code so the user can add the dependency.

Reply:
is it possible to have this as well as the default argument?

What I am trying to avoid is have `null` be passed in  for the tracer - from the app side it should just be `JaegerTracerPlugin()` not `JaegerTracerPlugin(null)` and if its null then instantiate the default tracer